nikola/plugins/command/check.py

```diff
@@ -273,14 +263,11 @@ def scan_links(self, find_sources=False, check_remote=False):
         self.logger.info("===============\n")
         self.logger.notice("{0} mode".format(self.site.config['URL_TYPE']))
         failure = False
-        for task in _call_nikola_list(self.l, self.site, ["--all"]):
-            task = task.strip()
-            if task.split(':')[0] in (
-                    'render_tags', 'render_archive',
-                    'render_galleries', 'render_indexes',
-                    'render_pages', 'render_posts',
-                    'render_site') and '.html' in task:
-                if self.analyze(task, find_sources, check_remote):
+        # Maybe we should just examine all HTML files
+        output_folder = self.site.config['OUTPUT_FOLDER']
+        for fname in _call_nikola_list(self.site):
+            if output_folder in fname and '.html' in fname:
```

Comment:
What about using `endswith()` here instead of (the second) `in`? Just in case someone likes to use `.html` in directory names :)